File: library/src/test/java/me/proxer/library/api/manga/ChapterEndpointTest.java
package me.proxer.library.api.manga;

import me.proxer.library.ProxerTest;
import me.proxer.library.api.ProxerException;
import me.proxer.library.entity.manga.Chapter;
import me.proxer.library.entity.manga.Page;
import me.proxer.library.enums.Language;
import okhttp3.mockwebserver.MockResponse;
import org.junit.Test;

import java.io.IOException;
import java.util.Arrays;
import java.util.Date;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Ruben Gees
 */
public class ChapterEndpointTest extends ProxerTest {

    @Test
    public void testDefault() throws IOException, ProxerException {
        server.enqueue(new MockResponse().setBody(fromResource("chapter.json")));

        final Chapter result = api.manga()
                .chapter("15", 1, Language.ENGLISH)
                .build()
                .execute();

        assertThat(result).isEqualTo(buildTestChapter());
    }

    @Test
    public void testPath() throws ProxerException, IOException, InterruptedException {
        server.enqueue(new MockResponse().setBody(fromResource("chapter.json")));

        api.manga().chapter("4523", 1, Language.GERMAN)
                .build()
                .execute();

        assertThat(server.takeRequest().getPath()).isEqualTo("/api/v1/manga/chapter?id=4523&episode=1&language=de");
    }

    private Chapter buildTestChapter() {
        return new Chapter("1954", "2784", "Chapter 1", "0", "nobody",
                new Date(1318716000L * 1000), null, null, "0", Arrays.asList(
                new Page("fairy-001-01.jpg", 1100, 765),
                new Page("fairy-001-02-03.jpg", 641, 900)
        ));
    }
}

File: library/src/test/kotlin/me/proxer/library/api/manga/ChapterEndpointTest.kt
package me.proxer.library.api.manga

import me.proxer.library.ProxerTest
import me.proxer.library.entity.manga.Chapter
import me.proxer.library.entity.manga.Page
import me.proxer.library.enums.Language
import me.proxer.library.runRequest
import org.amshove.kluent.shouldEqual
import org.junit.jupiter.api.Test
import java.util.Date

/**
 * @author Ruben Gees
 */
class ChapterEndpointTest : ProxerTest() {

    private val expectedChapter = Chapter(
        id = "1954", entryId = "2784", title = "Chapter 1", uploaderId = "0", uploaderName = "nobody",
        date = Date(1318716000L * 1000), scanGroupId = null, scanGroupName = null, server = "0",
        pages = listOf(
            Page(name = "fairy-001-01.jpg", height = 1100, width = 765),
            Page(name = "fairy-001-02-03.jpg", height = 641, width = 900)
        )
    )

    @Test
    fun testDefault() {
        val (result, _) = server.runRequest("chapter.json") {
            api.manga
                .chapter("15", 1, Language.ENGLISH)
                .build()
                .execute()
        }

        result shouldEqual expectedChapter
    }

    @Test
    fun testPath() {
        val (_, request) = server.runRequest("chapter.json") {
            api.manga.chapter("4523", 1, Language.GERMAN)
                .build()
                .execute()
        }

        request.path shouldEqual "/api/v1/manga/chapter?id=4523&episode=1&language=de"
    }
}
